nwbinspector/checks/tables.py

```diff
@@ -74,6 +75,39 @@ def check_time_intervals_stop_after_start(time_intervals: TimeIntervals, nelems:
         )
 
 
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=DynamicTable)
+def check_column_binary_capability(table: DynamicTable, nelems: int = 200):
+    """
+    Check each column of a table to see if the data could be set as a boolean dtype.
+
+    Parameters
+    ----------
+    time_intervals: TimeIntervals
+    nelems: int
+        Only check the first {nelems} elements. This is useful in case there columns are
+        very long so you don't need to load the entire array into memory. Use None to
+        load the entire arrays.
+    """
+    for column in table.columns:
+        if hasattr(column, "data"):
+            column_data = column.data if hasattr(column.data, "dtype") else np.array(column.data)
+            unique_values = np.unique(column_data[:nelems])
+            column_dtype = column_data.dtype
+            saved_bytes = (column_dtype.itemsize - 1) * column_data.size
```

Comment:
what is this doing?

Reply:
Reporting the # of bytes that could be saved if the user fixes this check

Reply:
Could we add a bytes formatter than naturally converts to KB, MB, GB, etc?

Reply:
Sure

Reply:
https://stackoverflow.com/a/1094933/2559070